x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_intra_app_state.tsx

```diff
@@ -27,11 +27,12 @@ export const IntraAppStateProvider = memo<{
   children: React.ReactNode;
 }>(({ kibanaScopedHistory, children }) => {
   const internalAppToAppState = useMemo<IntraAppState>(() => {
+    const intraAppState = kibanaScopedHistory.location.state as AnyIntraAppRouteState;
     return {
-      forRoute: kibanaScopedHistory.location.hash.substr(1),
-      routeState: kibanaScopedHistory.location.state as AnyIntraAppRouteState,
+      forRoute: intraAppState && intraAppState.baseRoute ? intraAppState.baseRoute : '',
```

Comment:
I'll defer to @jfsiii here, but personally I think we should parse the hash and remove the search params from it (if any), for a few reason:

1. This "bridge" solution here is temporary and should be looked at as a work-around to the fact that Hash router is still being used. So when this is removed (when Ingest moves to using BrowserRouter and the `ScoppedHistory `provided by Kibana's mount params), everything should just work normally (™️  famous words 😬 )
2. It feels wrong to have the Route interface state params define what `basePath` the state applies to, being that we're already redirecting to a specific location. I feel that the Route state params should be meant for the View that it displays, not for the intermediate process that bridges the two routing approaches
3. Adding this param in the state will further couple the provider with Ingest in having to know/define the `baseRoute` for the Route

Reply:
@jfsiii @paul-tavares my original apprehension to using `new URL()` is that I didn't see a way to get the `protocol` required to create the object without accessing `window.location` or just faking one.  I wasn't seeing it in the `kibanaScopedHistory` object anywhere.

After thinking about it, my understanding is that our goal here is to just use a library for URL parsing to be more safe, which makes complete sense.  So, I propose we just fake the protocol so that we're able to properly create the URL object and strip out the `pathname`

I'm doing this locally which is working right now.  As @paul-tavares said, we could back this out when we're no longer using a hashRouter.

```js
forRoute: (new URL(`https://localhost${kibanaScopedHistory.location.hash.substr(1)}`)).pathname
```

Reply:
@kevinlog  The [URL constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) has the signature

```javascript
const url = new URL(url [, base])
```

If we have a relative URL and are only after the pathname I think we can follow the example from

https://github.com/elastic/kibana/blob/7c22204b5a3db6caa449e21b4f48321112ff5609/x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/epm/hooks/use_links.tsx#L10-L11

e.g.
```javascript
const route = kibanaScopedHistory.location.hash.substr(1);
// url will have protocol, port, etc from second arg, but we're ignoring them
const url = new URL(route, 'http://any.tld.works'); 
```

Reply:
@jfsiii sounds good, I built the URL as above - it's all working as expected